scripts/commit_validation/commit_validation/validators/build_failure_rca_validator.py

```diff
@@ -0,0 +1,59 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+
+import fnmatch
+import json
+import re
+from typing import Type, List
+
+from commit_validation.commit_validation import Commit, CommitValidator, EXCLUDED_VALIDATION_PATTERNS, VERBOSE
+
+RCA_PATTERN_PATH = '*/scripts/build/build_failure_rca/rca_patterns/*'
+
+class BuildFailureRCAValidator(CommitValidator):
+    """A file-level validator that makes sure build failure RCA patterns can catch all the test cases"""
+
+    def run(self, commit: Commit, errors: List[str]) -> bool:
+        for file_name in commit.get_files():
+            for pattern in EXCLUDED_VALIDATION_PATTERNS:
+                if fnmatch.fnmatch(file_name, pattern):
+                    if VERBOSE: print(f'{file_name}::{self.__class__.__name__} SKIPPED - Validation pattern excluded on path.')
+                    break
+            else:
+                if fnmatch.fnmatch(file_name, RCA_PATTERN_PATH):
```

Comment:
Should this be `if not` instead?

Reply:
Good catch! Not sure how I missed "not"...